.github/workflows/unit.yaml

```diff
@@ -0,0 +1,57 @@
+name: Unit
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  unit:
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.allow_failure }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - ruby: 2.4.9
+          puppet: 5
+          fixtures: .fixtures.yml
+          allow_failure: false
+        - ruby: 2.5.7
+          puppet: 6
+          fixtures: .fixtures.yml
+          allow_failure: false
+        - ruby: 2.7.0
+          puppet: 7
+          fixtures: .fixtures.yml
+          allow_failure: false
+        - ruby: 2.4.9
```

Comment:
This will have to be changed most likely, and I can add to DESIGN.md with other documentation to acceptance testing.  The issue per other comments is there is no easy way to mark certain items in the matrix as allowed to fail.

Reply:
Added a comment about why there is allow_failure and what it does.